### PR TITLE
HPCC-17846 DOCS:Additional Info for Multiple Thors

### PR DESCRIPTION
**Pull request by g-pan**
Fix HPCC-17846 DOCS:Additional Info for Multiple Thors
Adding additional information to the section about running multiple Thors
on the same cluster in the System Administrators book.

@jakesmith please review this content as it was taken from an email thread of a user's question.
@JamesDeFabia  please review

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

Test results (and PDF) can be viewed at http://10.240.32.243/view/Docs/job/DocDev0V/17/CE=ubuntu-14.04-amd64-1/  Systems Administrator's Guide Page 121

### CODE REVIEW

File: docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml [hunk 1]
```diff
@@ -1662,8 +1662,15 @@ dfsSSLPrivateKeyFile=/keyfilepath/keyfile</programlisting>Set the <emphasis
 
         <para>The downside of running multiple Thors on the same hardware is
         that the physical memory on the nodes needs to be shared among each of
-        the Thors. This needs to be configured per Thor cluster
-        definition.</para>
+        the Thors. This needs to be configured per Thor cluster definition.
+        </para>
+
+        <para>Multiple Thor's on the same cluster require them to share the
```
Comment (JamesDeFabia) [inline docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml]: No apostrophe on plural form of Thor.

File: docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml [hunk 2]
```diff
@@ -1662,8 +1662,15 @@ dfsSSLPrivateKeyFile=/keyfilepath/keyfile</programlisting>Set the <emphasis
...
+        same build and installation. The environment defines each Thor
+        cluster, which can share the same machine set. There are slave and
+        master port settings that need to be set to avoid clashing. There are
+        also memory sharing/splitting considerations and settings that need to
+        be made. </para>
```
Comment (JamesDeFabia) [inline docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml]: You say there are settings to be set to avoid clashing but don't say which ones. You also say there are memory considerations but don't say what those considerations are.

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
One grammar typo plus some details that need further explanation.

**Review by JamesDeFabia — COMMENTED** [review]
Looks good from my POV, but @jakesmith needs to have the last word

**Comment by richardkchapman** [comment]
@jakesmith

File: docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml [hunk 3]
```diff
@@ -1662,8 +1662,93 @@ dfsSSLPrivateKeyFile=/keyfilepath/keyfile</programlisting>Set the <emphasis
...
+                  <entry>This value is the increment from the base slave port.
+                  </entry>
+                </row>
+
+                <row>
+                  <entry><emphasis
+                  role="bold">masterMemorySize</emphasis></entry>
+
+                  <entry>The maximum memory a Thor master can use. If left
+                  blank it will use the global value.</entry>
```
Comment (jakesmith) [inline docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml]: perhaps should say : ' If left blank it will use the globalMemorySize value. '

File: docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml [hunk 4]
```diff
@@ -1662,8 +1662,93 @@ dfsSSLPrivateKeyFile=/keyfilepath/keyfile</programlisting>Set the <emphasis
...
+                  <entry>The name of each Thor instance must be
+                  unique.</entry>
+                </row>
+
+                <row>
+                  <entry><emphasis role="bold">nodeGroup</emphasis></entry>
+
+                  <entry>This value determines where the sprayed data goes and
+                  it must be the same for the same set of nodes (IP
+                  addresses).</entry>
```
Comment (jakesmith) [inline docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml]: I think I'd reword this as:

The nodeGroup is associated with files published by this Thor instance, normally it is left blank and defaults to the same as the 'name' attribute. In environments with multiple Thor's sharing the same group of nodes, the 'name' of each Thor must be different, but the nodeGroup value of all the Thor's sharing the same physical nodes should be set to the same name. NB: It is important to make the nodeGroup value equal to one of the Thor instance name values.

**Comment by jakesmith** [comment]
@g-pan - a couple of comments

**Comment by g-pan** [comment]
Updated to include @jakesmith comments. Thank you.

**Comment by jakesmith** [comment]
@g-pan - looks fine.

@richardkchapman - ready to merge.
